include/nanorange/view/empty.hpp

```diff
@@ -0,0 +1,51 @@
+// nanorange/view/empty.hpp
+//
+// Copyright (c) 2018 Tristan Brindle (tcbrindle at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef NANORANGE_VIEW_EMPTY_HPP_INCLUDED
+#define NANORANGE_VIEW_EMPTY_HPP_INCLUDED
+
+#include <nanorange/view/interface.hpp>
+
+NANO_BEGIN_NAMESPACE
+
+namespace empty_view_ {
```

Comment:
Why the namespace?

Reply:
Because Clang is silly. It's already seen the definition of `nano::begin` as an object, so it complains about declaring another `begin` as a friend function of a class in the same namespace, even though the two don't conflict. Putting things in an inner namespace and then bringing them into the parent namespace with `using` is a workaround -- `common_iterator` and `subrange` need to do the same thing, and maybe others that I've forgotten.